src/relations/vault.py

```diff
@@ -105,11 +120,47 @@ def _get_vault_config(self):
         role_id = secret_content["role-id"]
         role_secret_id = secret_content["role-secret-id"]
 
+        certs_path = self.get_ca_cert_location_in_charm()
+        with open(f"{certs_path}/{VAULT_CA_CERT_FILENAME}", "w") as fd:
+            fd.write(ca_certificate)
+
         return {
-            "TWC_VAULT_ADDR": vault_url,
-            "TWC_VAULT_CACERT_BYTES": ca_certificate,
-            "TWC_VAULT_ROLE_ID": role_id,
-            "TWC_VAULT_ROLE_SECRET_ID": role_secret_id,
-            "TWC_VAULT_MOUNT": mount,
-            "TWC_VAULT_CERT_PATH": VAULT_CERT_PATH,
+            "vault_address": vault_url,
+            "vault_ca_certificate_bytes": ca_certificate,
+            "vault_role_id": role_id,
+            "vault_role_secret_id": role_secret_id,
+            "vault_mount": mount,
+            "vault_cert_path": VAULT_CERT_PATH,
         }
+
+    def get_vault_client(self):
+        """Initialize Vault client.
+
+        Returns:
+            Vault client.
+        """
+        ca_certificate_path = self.get_ca_cert_location_in_charm()
+        vault_config = self.get_vault_config()
+        return VaultClient(
+            address=vault_config["vault_address"],
+            cert_path=f"{ca_certificate_path}/{VAULT_CA_CERT_FILENAME}",
+            role_id=vault_config["vault_role_id"],
+            role_secret_id=vault_config["vault_role_secret_id"],
+            mount_point=vault_config["vault_mount"],
+        )
```

Comment:
[Suggestion] Here you can use spread annotation like the below:

```
        return {
            "address": vault_url,
            "role_id": role_id,
            "role_secret_id": role_secret_id,
            "mount": mount,
        }

    def get_vault_client(self):
        """Initialize Vault client.

        Returns:
            Vault client.
        """
        ca_certificate_path = self.get_ca_cert_location_in_charm()
        vault_config = self.get_vault_config()
        return VaultClient(
            **{**vault_config, 'cert_path': f"{ca_certificate_path}/{VAULT_CA_CERT_FILENAME}"}
        )
```

`vault_ca_certificate_bytes` and `vault_cert_path` are not used anywhere so you can remove them from return dictionary.